stanford/roles/deployment/tasks/main.yml

```diff
@@ -94,9 +94,14 @@
     region: "{{ DEPLOYMENT_REGION }}"
     if_exist_do_not_create: true
   register: new_nat_gateway
+  when: DEPLOYMENT_NAT_GATEWAYS is not defined
 - debug: var=new_nat_gateway
 - name: Set NAT gateway facts
   set_fact:
     DEPLOYMENT_NAT_GATEWAY_ID: "{{ new_nat_gateway.nat_gateway_id  }}"
-  when: DEPLOYMENT_NAT_GATEWAY_ID is not defined and DEPLOYMENT_SUBNET_PUBLIC_IDS is defined
```

Comment:
I caught this bug just prior to merging.
- This was a copy/paste failure. We shouldn't be checking `DEPLOYMENT_SUBNET_PUBLIC_IDS`; that's irrelevant. If we need a NAT gateway, we should create it, whether or not we just created the subnet.
- And since there are two nat gateways (one per AZ), we need to do this like it's done above w/ the subnets, _not_ like the internet gateway, which is to say:
  - optionally store a list of values. if the list is not defined, create one and set the ID to it. If the list _is_ defined, just grab the correct one, based on the `deployment_subnet_index`.